src/string_search.h

```diff
@@ -0,0 +1,597 @@
+// Copyright 2011 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SRC_STRING_SEARCH_H_
+#define SRC_STRING_SEARCH_H_
+
+#include "node.h"
+#include <string.h>
+
+namespace node {
+namespace stringsearch {
+
+// Returns the maximum of the two parameters.
+template <typename T> T Max(T a, T b) { return a < b ? b : a; }
+
+static const uint32_t kMaxOneByteCharCodeU = 0xff;
+
+static inline int NonOneByteStart(const uint16_t *chars, int length) {
+  const uint16_t *limit = chars + length;
+  const uint16_t *start = chars;
```

Comment:
Nah Chromium style is based on the base Google style but is different on the same things you want:
````
ChromiumStyle.AllowAllParametersOfDeclarationOnNextLine = false;
ChromiumStyle.AllowShortFunctionsOnASingleLine = FormatStyle::SFS_Inline;
ChromiumStyle.AllowShortIfStatementsOnASingleLine = false;
ChromiumStyle.AllowShortLoopsOnASingleLine = false;
ChromiumStyle.BinPackParameters = false;
ChromiumStyle.DerivePointerAlignment = false;
````

And the chromium style guide explicitly for example specifies:

Use size_t for object and allocation sizes, object counts, array and pointer offsets, vector indices, and so on. The signed types are incorrect and unsafe for these purposes. (Integer overflow behavior for signed types is undefined in the C and C++ standards, while the behavior is defined for unsigned types.) The C++ STL is a guide here: they use size_t and foo::size_type for very good reasons.